modules/navier_stokes/src/fv/bcs/FVPressurePin.C

```diff
@@ -0,0 +1,162 @@
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVPressurePin.h"
+#include "MooseMesh.h"
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "libmesh/node.h"
+#include "libmesh/elem.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/numeric_vector.h"
+
+registerMooseObject("MooseApp", FVPressurePin);
+
+defineLegacyParams(FVPressurePin);
+
+InputParameters
+FVPressurePin::validParams()
+{
+  InputParameters params = NodalBCBase::validParams();
+  params.addRequiredParam<Real>("value", "Value of the BC");
+  params.declareControllable("value");
+  params.addClassDescription("Imposes a pressure pin constraint.");
+  return params;
+}
+
+void
+FVPressurePin::boundaryParamError()
+{
+  paramError("boundary",
+             "The FVPressurePin class should only be associated with a single boundary node.");
+}
+
+FVPressurePin::FVPressurePin(const InputParameters & parameters)
+  : NodalBCBase(parameters),
+    MooseVariableInterface<Real>(this,
+                                 false,
+                                 "variable",
+                                 Moose::VarKindType::VAR_NONLINEAR,
+                                 Moose::VarFieldType::VAR_FIELD_STANDARD),
+    _var(*mooseVariableFV()),
+    _fe_problem(*getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")),
+    _sys(*getCheckedPointerParam<SystemBase *>("_sys")),
+    _mesh(getCheckedPointerParam<SubProblem *>("_subproblem")->mesh()),
+    _value(getParam<Real>("value"))
+{
+  addMooseVariableDependency(&_var);
+
+  const auto & boundary_names = getParam<std::vector<BoundaryName>>("boundary");
+
+  if (boundary_names.size() != 1)
+    boundaryParamError();
+
+  const auto boundary_id = _mesh.getBoundaryID(boundary_names[0]);
+
+  const auto & nodeset_to_nodes = _mesh.nodeSetNodes();
+
+  auto it = nodeset_to_nodes.find(boundary_id);
+
+  const std::vector<dof_id_type> * nodes = nullptr;
+  bool local_node = false;
+
+  if (it != nodeset_to_nodes.end())
+    nodes = &it->second;
+
+  if (nodes && nodes->size() > 1)
+    paramError("boundary",
+               "The FVPressurePin class should only be associated with a single boundary node.");
+
+  const Node * const one_node = nodes ? _mesh.getMesh().query_node_ptr((*nodes)[0]) : nullptr;
+  mooseAssert(!nodes || one_node,
+              "It's weird that this nodeid existed in our map, but we can't query it on the mesh.");
+  mooseAssert(!nodes || one_node->id() == (*nodes)[0], "the ids should match");
+
+  local_node = nodes && one_node->processor_id() == this->processor_id();
+
+  std::vector<dof_id_type> global_nodes;
+  if (local_node)
+    global_nodes.push_back((*nodes)[0]);
+
+  _communicator.gather(0, global_nodes);
+
+  if (_communicator.rank() == 0)
+    if (global_nodes.size() != 1)
+      boundaryParamError();
+
+  // lindsayad: No matter how we refine, this node should still exist and be executed on (right?)
```

Comment:
Hmm this is pretty tricky. This is so much easier to implement with a nodal basis because a node on the base mesh exists on a refined mesh no matter how it is refined. It is indeed not ideal to change the solution profile as the mesh is refined. I'm going to have to mull this over. It might be best to do a global constraint for this like in https://github.com/idaholab/moose/issues/15549

Reply:
Ok, I implemented `FVScalarLagrangeMultiplier` which is a much more fool-proof way of removing the nullspace. I've dropped `FVPressurePin` completely